cairo/tests/fixtures/compiler.py

```diff
@@ -40,20 +41,25 @@ def cairo_file(request):
     return cairo_file
 
 
-@pytest.fixture(scope="module")
-def cairo_program(cairo_file):
-    start = perf_counter()
-    program = cairo_compile(cairo_file)
-    program.hints = implement_hints(program)
-    stop = perf_counter()
-    logger.info(f"{cairo_file} compiled in {stop - start:.2f}s")
-    return program
-
-
 @pytest.fixture(scope="module")
 def main_path(cairo_file):
     """
     Resolve the __main__ part of the cairo scope path.
     """
     parts = cairo_file.relative_to(Path.cwd()).with_suffix("").parts
     return parts[1:] if parts[0] == "cairo" else parts
+
+
+@pytest.fixture(scope="module")
+def cairo_program(cairo_file, main_path):
+    start = perf_counter()
+    program = cairo_compile(cairo_file)
+    program.hints = implement_hints(program)
+    all_identifiers = list(program.identifiers.dict.items())
+    for k, v in all_identifiers:
```

Comment:
when running the tests, the main file is the test file and the compiler is not able to find struct defined therein in a breakpoint, looking for, e.g. `ethereum.cancun.trie.InternalNode` while only `__main__.InternalNode` exists. There is probably a better way to solve this at the IdentifierManager level, but it's not part of this repo and I'm not sure how fast can PR be merged on cairo-lang. Consequently, I've just decided to add them "manually"

Reply:
Add as comment for postériorité